services/ui-src/src/components/reports/ReportProvider.tsx

```diff
@@ -171,6 +176,13 @@ export const ReportProvider = ({ children }: Props) => {
     );
   };
 
+  useEffect(() => {
+    const submittedReports = reportsByState?.filter(
+      (item) => item.status === "Submitted"
+    );
+    setSubmittedReportsByState(submittedReports);
+  }, [reportsByState]);
```

Comment:
Does this need to be its own slice of state when it is filtered off another slice of state? I think this conceptually is fine here with react state, but might flag as an antipattern with redux.

See second comment around use case

Reply:
I like it better here for reasons in other comment :)